addons/account/models/account_account.py

```diff
@@ -147,38 +147,45 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
         if fname == 'code':
             return self.with_company(self.env.company.root_id).sudo()._field_to_sql(alias, 'code_store', query, flush)
         if fname == 'placeholder_code':
+            # The placeholder_code is defined as the account's code in the first active company to
+            # which the account belongs.
             query.add_join(
-                'JOIN',
+                'LEFT JOIN',
                 'account_first_company',
                 SQL(
-                    """(
-                        SELECT DISTINCT ON (rel.account_account_id)
-                               rel.account_account_id AS account_id,
-                               rel.res_company_id AS company_id,
-                               SPLIT_PART(res_company.parent_path, '/', 1) AS root_company_id,
-                               res_company.name AS company_name
-                          FROM account_account_res_company_rel rel
-                          JOIN res_company
-                            ON res_company.id = rel.res_company_id
-                         WHERE rel.res_company_id IN %(authorized_company_ids)s
-                      ORDER BY rel.account_account_id, company_id
+                    """LATERAL (
+                        SELECT DISTINCT ON (account_companies.account_account_id)
+                               account_companies.res_company_id AS company_id,
+                               SPLIT_PART(active_company.parent_path, '/', 1) AS root_company_id,
+                               active_company.name AS company_name
+                          FROM account_account_res_company_rel account_companies
+                          JOIN unnest(%(authorized_company_ids)s) WITH ORDINALITY AS active_company_ordering(id, idx)
+                            ON active_company_ordering.id = account_companies.res_company_id
+                          JOIN res_company active_company
+                            ON active_company.id = active_company_ordering.id
+                         WHERE account_companies.account_account_id = %(account_id)s
+                      ORDER BY account_companies.account_account_id,
+                               active_company_ordering.idx
```

Comment:
We already take the right order by taking `active_company_root_id` in the `COALESCE`

Reply:
Sure, I'll just keep it as it currently is.